crates/ruff/src/checkers/physical_lines.rs

```diff
@@ -154,6 +160,24 @@ pub(crate) fn check_physical_lines(
                 diagnostics.push(diagnostic);
             }
         }
+        if enforce_copyright_header && !checked_copyright_header_absent {
```

Comment:
Could also do dep injection here and pass regexp as an arg. However, that would essentially put all the logic in physical_lines.rs. Let me know which you would prefer